lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -302,10 +302,10 @@ def write(chunk)
           unless label_map.nil?
             labels = {}
             @label_map.each do |field, label|
-              if record.key?(field)
-                labels[label] = record[field]
-                record.delete(field)
-              end
+              next unless record.key?(field)
+              # label values are required to be strings.
+              labels[label] = record[field].to_s
```

Comment:
Does "to_s" work on every type? 
What happens when this field is a record (hash?) itself?